src/tiledb/cloud/bioimg/ingestion.py

```diff
@@ -34,9 +41,35 @@ def ingest(
         defaults to None
     :param namespace: The namespace where the DAG will run
     """
+    source_list = [source] if isinstance(source, str) else list(source)
+    max_workers = None if num_batches else 20  # Default picked heuristically.
+
+    def build_io_uris() -> List[Tuple[str, str]]:
+        """Match input uri/s with output destinations
+
+        :param source: A sequence of paths or path to input
+        :param output_dir: A path to the output directory
+        """
+        vfs = tiledb.VFS(config=config)
+        my_source_list = source_list
+        if len(my_source_list) == 1 and vfs.is_dir(my_source_list[0]):
+            # Folder like input
+            my_source_list = vfs.ls(my_source_list[0])
+        return [
+            (uri, os.path.join(output, posixpath.basename(uri)))
+            for uri in my_source_list
+        ]
+
+    def build_input_batches():
+        """Groups input URIs into batches."""
+        uri_pairs = build_io_uris()
+        # If the user didn't specify a number of batches, run every import
+        # as its own task.
+        my_num_batches = num_batches or len(uri_pairs)
+        return [uri_pairs[n::num_batches] for n in range(my_num_batches)]
```

Comment:
If the `num_batches` given by the user is greater than the `uri_pairs` length. We get empty lists like:
```python
pairs = [1,2,3,4,5,6,7,8,9,10]
num_batches = 20
[pairs[n::num_batches] for n in range(num_batches)]
Out[6]: 
[[1], [2], [3], [4], [5], [6], [7], [8], [9], [10], [], [], [], [], [], [], [], [], [], []]
```
Will this create empty tasks for the lister?

Reply:
It would. Fixed.